base/uk.ac.stfc.isis.ibex.opis/resources/jaws/blades_script.py

```diff
@@ -52,23 +66,69 @@ def run(self):
                 return
 
             self.get_pvs()
-
+            
+            #Without the following code, if the horizontal and vertical limits of travel differ
+            #but the vertical and horizontal gaps are set equal, the gap on the GUI will not appear
+            #square as expected. The following applies a scaling factor to the jaw heights/widths to correct this
+            #and calculates x and y positions for the limit of travel indicators.
+            if self.max_x > self.max_y:
+                scaling_factor = self.max_y/self.max_x
+                self.north *= scaling_factor
+                self.south *= scaling_factor
+                north_bound_val = background_height*scaling_factor/2.0
+                south_bound_val = background_height - background_height*scaling_factor/2.0
+            elif self.max_y > self.max_x:
+                scaling_factor = self.max_x/self.max_y
+                self.east *= scaling_factor
+                self.west *= scaling_factor
+                west_bound_val = background_width*scaling_factor/2.0
+                east_bound_val = background_width - background_width*scaling_factor/2.0
+            else:
+                scaling_factor = 1.0             
+                        
             north_height, south_height = self.calc_pair_height((self.north, self.south), self.max_y, background_height)
             east_width, west_width = self.calc_pair_height((self.east, self.west), self.max_x, background_width)
-
+            
             south_y = background_y + background_height - south_height
             east_x = background_x + background_width - east_width
-
-            north_blade.setPropertyValue("height", north_height)
-            south_blade.setPropertyValue("height", south_height)
-            south_blade.setPropertyValue("y", south_y)
-
-            west_blade.setPropertyValue("width", west_width)
-            east_blade.setPropertyValue("width", east_width)
-            east_blade.setPropertyValue("x", east_x)
-
+            
+            #Make GUI changes on the GUI thread only
+            max_x = self.max_x
+            max_y = self.max_y
+            class UITask(Runnable):
+                def run(self):
+                    #Set limit of travel indicators
+                    if max_x > max_y:
+                        north_bound.setPropertyValue("visible", True)
+                        south_bound.setPropertyValue("visible", True) 
+                        east_bound.setPropertyValue("visible", False)
+                        west_bound.setPropertyValue("visible", False)                       
+                        north_bound.setPropertyValue("y", north_bound_val)
+                        south_bound.setPropertyValue("y", south_bound_val)
```

Comment:
I think we can just always set the y/x of `north_bound`, `south_bound`, `east_bound` and `west_bound` regardless of if they're visible or not. The additional overhead is minimal and it tidies the code up a bit.